src/unix/fs.c

```diff
@@ -1148,8 +1148,19 @@ static ssize_t uv__fs_copyfile(uv_fs_t* req) {
 
   dst_flags = O_WRONLY | O_CREAT | O_TRUNC;
```

Comment:
Would it possibly make more sense to remove the `O_TRUNC` flag from here? I'm not sure whether there would be a TOCTOU concern now with using `stat` on the path instead of the file descriptor.

Reply:
> Would it possibly make more sense to remove the O_TRUNC flag from here?

Wouldn't that break copy operations that overwrite existing files?

> I'm not sure whether there would be a TOCTOU concern now with using stat on the path instead of the file descriptor.

I thought about that. Since we're only concerned about the case where the source and destination are the same file (and the source is already opened), I couldn't think of a meaningful TOCTOU.

Reply:
> > Would it possibly make more sense to remove the O_TRUNC flag from here?
> 
> Wouldn't that break copy operations that overwrite existing files?

We could manually truncate using `ftruncate()` after `fstat()` on the already-opened fd, right?

Reply:
I think that would work. But, unless there is a real issue with the current approach in this PR, wouldn't that be more work in most cases? Currently, in this PR there is a `stat()` call if `UV_FS_COPYFILE_EXCL` is not present. The alternative would be unconditional `ftruncate()` and `fstat()` calls (assuming the `open()` succeeded).

Reply:
I'd guess that `fstat` is a lot less work for the kernel than `stat` (one involves a long directory walk and requires checking permissions, the other doesn't), so the `ftruncate` would be cheaper? It seems also easy enough to add the `O_TRUNC` flag in the `UV_FS_COPYFILE_EXCL` branch?

Reply:
Updated to use ftruncate.